UserAgeCalculation

```diff
@@ -9,22 +9,52 @@ namespace UserAgeCalculation
             Console.Write("Enter your name: ");
             string userName = Console.ReadLine();
 
+            int currentYear = DateTime.Now.Year;
 
-            Console.Write("Enter your year of birth: ");
-            int yearOfBirth = Convert.ToInt32(Console.ReadLine());
+            int yearOfBirth;
 
-            int currentYear = DateTime.Now.Year;
-            int userAge = currentYear - yearOfBirth; 
+            while (true)
+            {
+                Console.Write("Enter your year of birth: ");
+                string yearOfBirthString = Console.ReadLine();
+
+                bool yearOfBirthBool = int.TryParse(yearOfBirthString, out yearOfBirth);
+
+                if (yearOfBirth > 0)
```

Comment:
use more complex if statements to reduce nested if-s. For example : 

`if ( 0 < yearOfBirth <= (currentYear - 100) )`